### PR TITLE
Release notes for the Operator for KC 20.

### LINKED ISSUE #14881: Release notes for the Operator for KC 20
* Announce Operator no longer preview
* Document breaking changes introduced in:
   * https://github.com/keycloak/keycloak/issues/14410
   * https://github.com/keycloak/keycloak/issues/14407
   * https://github.com/keycloak/keycloak/issues/14396
   * https://github.com/keycloak/keycloak/issues/14395

### PR DESCRIPTION
**Pull request by andre-nascimento6791**
- Some paragraphs added to "/release_notes/topics/20_0_0.adoc";

- Some paragraphs added to "/upgrading/topics/keycloak/changes-20_0_0.adoc";

Closes https://github.com/keycloak/keycloak/issues/14881

### CODE REVIEW

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 1]
```diff
@@ -27,4 +27,61 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
 
 H2 database base files created with H2 version 1.x should not be used with version 2.x.
 
-Purge existing H2 database files to start with an empty database, export and import the realms using Keycloak's export and import functionality, or refer to the http://www.h2database.com/html/migration-to-v2.html[migration notes on the H2 database project's website] for details on how to migrate H2 database contents.
\ No newline at end of file
+Purge existing H2 database files to start with an empty database, export and import the realms using Keycloak's export and import functionality, or refer to the http://www.h2database.com/html/migration-to-v2.html[migration notes on the H2 database project's website] for details on how to migrate H2 database contents.
+
+= Breaking changes in the new version of Keycloak Operator
+
+NOTE: In order to use the newest version of the Keycloak Operator, the manual reinstallation and upgrade of your CRs is required. There’s no automated migration.
```
Comment (Pepo48) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: *are required

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 2]
```diff
@@ -27,4 +27,61 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
+
+This release contains following breaking changes in Keycloak CRs:
```
Comment (Pepo48) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: *the following

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 3]
```diff
@@ -27,4 +27,61 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
+
+== serverConfiguration free-form field was renamed
+
+From now on it’s called `additionalOptions`. The idea behind this decision is to align it more with the Keycloak Quarkus distribution and achieve/preserve a naming consistency.
+`serverConfiguration` can be still used for configuring options that haven’t got a declared alternative in Keycloak custom resources (CRs). A good example of such use can be service providers.
+
+== Ingress options were refined
+
+In the past it used to be defined via `disableDefaultIngress` boolean property. We decided to clarify it a bit, thus from now on you can use the following structure to control your ingress settings:
```
Comment (Pepo48) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: *in the past,

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 4]
```diff
@@ -27,4 +27,61 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
+```yaml
+spec:
+    ...
+    ingress:
+      enabled: false
+```
+
+== HTTP options were added
+
+Similarly, like with ingress, you can define multiple HTTP option in a structuralized manner:
```
Comment (Pepo48) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: *options

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 3 above)
Comment (Pepo48) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: *via the disableDefaultIngress property

File: release_notes/topics/20_0_0.adoc [hunk 5]
```diff
@@ -17,3 +17,9 @@ In this release, we are making important changes to `kc.bat` to give the same ex
 
 In this release, the H2 driver has been upgraded from version 1.x to version 2.x.
 See link:{upgradingguide_link}[{upgradingguide_name}] for details on how to migrate.
+
+= Keycloak Operator upgrade
+
+We are happy to announce that Keycloak Operator is no longer a tech preview feature. We added new functionality as well as quality of life improvements. Some of them resulted to breaking changes.
```
Comment (Pepo48) [inline release_notes/topics/20_0_0.adoc]: *the quality of life

Comment (Pepo48) [inline release_notes/topics/20_0_0.adoc]: *resulted in

**Review by Pepo48 — CHANGES_REQUESTED** [review]
I corrected some of the mistakes I found.

**Review by vmuzikar — COMMENTED** [review]
Changes LGTM (apart from what @Pepo48 already mentioned).

On top of that I realized that we should probably also mention that we are removing the `INSECURE-DISABLE` special value for `hostname` and `tlsSecret`. Please see the [docs PR](https://github.com/keycloak/keycloak/pull/15088/files#diff-406ef71b320dcd3e6bbbe2fdbb7d4c233fa10d6fbc3e26df83d7d6142f715243).

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 1 above)
Comment (andre-nascimento6791) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Fixed. Done!.
Thanks.

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 2 above)
Comment (andre-nascimento6791) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Fixed. Done!.
Thanks.

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 3 above)
Comment (andre-nascimento6791) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Fixed. Done!.
Thanks.

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 4 above)
Comment (andre-nascimento6791) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Fixed. Done!.
Thanks.

File: release_notes/topics/20_0_0.adoc (hunk 5 above)
Comment (andre-nascimento6791) [reply release_notes/topics/20_0_0.adoc]: Fixed. Done!.
Thanks.

Comment (andre-nascimento6791) [reply release_notes/topics/20_0_0.adoc]: Fixed. Done!.
Thanks.

**Comment by andre-nascimento6791** [comment]
> Changes LGTM (apart from what @Pepo48 already mentioned).
> 
> On top of that I realized that we should probably also mention that we are removing the `INSECURE-DISABLE` special value for `hostname` and `tlsSecret`. Please see the [docs PR](https://github.com/keycloak/keycloak/pull/15088/files#diff-406ef71b320dcd3e6bbbe2fdbb7d4c233fa10d6fbc3e26df83d7d6142f715243).

Added the requested mention.
Thanks.

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 6]
```diff
@@ -27,4 +27,63 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
+== Hostname options were added
+
+Last but not least, the hostname options were also changed:
+
+```yaml
+spec:
+    ...
+  hostname:
+    hostname: [keycloak-server-hostname]
+    hostnameUrl: [url]
```
Comment (Pepo48) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: This property was omitted in the end. Please, remove it.

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 7]
```diff
@@ -27,4 +27,63 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
+    admin: [admin-console-hostname]
+    adminUrl: [admin-console-base-url]
+    port: [port number]
```
Comment (Pepo48) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: This property was omitted in the end. Please, remove it.

**Review by Pepo48 — CHANGES_REQUESTED** [review]
These two properties need to be removed.

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 8]
```diff
@@ -27,4 +27,63 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
+    strict: [true|false]
+    strictBackchannel: [true|false]
+```
+
+NOTE: There was removed the possibility to set `INSECURE-DISABLE` special value to `hostname` and `tlsSecret` fields.
```
Comment (vmuzikar) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: Maybe I'd rephrase a little bit.
```suggestion
== Some fields are no longer required

The `hostname` and `tlsSecret` fields are now optional to align with the Quarkus distribution configuration. With that we also removed the possibility to set `INSECURE-DISABLE` special value to those fields. In order to disable hostname checks and enable HTTP, please follow the same approach as with the Quarkus distribution, i.e. set `strict: false`, `strictBackchannel: false` and `httpEnabled: true` fields.
```

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 6 above)
Comment (andre-nascimento6791) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Done.
Thanks.

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 7 above)
Comment (andre-nascimento6791) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Done.
Thanks.

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 8 above)
Comment (andre-nascimento6791) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Done.
Thanks.

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 3 above)
Comment (andre-nascimento6791) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: Done.
Thanks.

**Review by vmuzikar — DISMISSED** [review]
LGTM, thank you.

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 9]
```diff
@@ -27,4 +27,64 @@ See the H2 documentation http://www.h2database.com/html/commands.html#set_non_ke
...
+```
+
+== Some fields are no longer required
+
+The `hostname` and `tlsSecret` fields are now optional to align with the Quarkus distribution configuration. With that we also removed the possibility to set `INSECURE-DISABLE` special value to those fields. In order to disable hostname checks and enable HTTP, please follow the same approach as with the Quarkus distribution, i.e. set `strict: false`, `strictBackchannel: false` and `httpEnabled: true` fields.
+
+
+= OLM channel was changed to fast
+
+The default channel of Keycloak Operator Lifecycle Manager changed to `fast`.
```
Comment (Pepo48) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]: @andre-nascimento-rh (cc: @vmuzikar) maybe just one final detail - add was/is before the changed verb?

Otherwise LGTM.

**Review by Pepo48 — DISMISSED** [review]

File: upgrading/topics/keycloak/changes-20_0_0.adoc (hunk 9 above)
Comment (vmuzikar) [reply upgrading/topics/keycloak/changes-20_0_0.adoc]: @andre-nascimento-rh I added a commit to address Peter's comment, if that's ok.

**Review by vmuzikar — APPROVED** [review]

**Review by Pepo48 — APPROVED** [review]
